application/controllers/admin/Tokens.php

```diff
@@ -680,24 +680,29 @@ public function edit($iSurveyId, $iTokenId, $ajax = false)
             }
 
             // completed
-            if (trim($request->getPost('completed')) === 'N') {
+            if (trim($request->getPost('completed')) == 'N' || trim($request->getPost('completed')) == '') {
                 $_POST['completed'] = 'N';
-            } else {
+            } elseif (trim($request->getPost('completed')) == 'Y') {
                 $_POST['completed'] = 'Y';
+            } else {
+                $datetimeobj = new Date_Time_Converter(trim($request->getPost('completed')), $dateformatdetails['phpdate'] . ' H:i');
+                $_POST['completed'] = $datetimeobj->convert('Y-m-d H:i');
             }
 
             //sent
-            if (trim($request->getPost('sent')) === 'N') {
+            if (trim($request->getPost('sent')) == 'N') {
                 $_POST['sent'] = 'N';
             } else {
-                 $_POST['sent'] = 'Y';
+                $datetimeobj = new Date_Time_Converter(trim($request->getPost('sent')), $dateformatdetails['phpdate'] . ' H:i');
+                $_POST['sent'] = $datetimeobj->convert('Y-m-d H:i');
```

Comment:
invalid date set to false . Need to check data validity

Reply:
This is just like it is on LTS.
If there is a new bug, maybe we should just create a new ticket.
I know is not your preferred way, but it is hard to streamline the process if not.